executor/src/witgen/jit/witgen_inference.rs

```diff
@@ -462,18 +498,20 @@ impl<'a, T: FieldElement, FixedEval: FixedEvaluator<T>> Evaluator<'a, T, FixedEv
 
 /// An equality constraint between an algebraic expression evaluated
 /// on a certain row offset and a variable or fixed constant value.
+#[derive(Clone)]
 struct Assignment<'a, T: FieldElement> {
     lhs: &'a Expression<T>,
     row_offset: i32,
     rhs: VariableOrValue<T, Variable>,
 }
 
+#[derive(Clone)]
 enum VariableOrValue<T, V> {
     Variable(V),
     Value(T),
 }
 
-pub trait FixedEvaluator<T: FieldElement> {
+pub trait FixedEvaluator<T: FieldElement>: Clone {
```

Comment:
In practice, we just pass a reference, which implements `Clone`, right? I guess the advantage of this is that we can also pass something like `Rc<_>`, and it can automatically free the memory?

I guess it's the same for `CanProcessCall` below, which is not restricted to be `Clone` as well, but it is in practice. I feel like accepting references and not requiring `Clone` would reduce verbosity, but no strong opinion...

Reply:
I'd like to avoid the lifetimes...